model-integration/src/main/java/ai/vespa/embedding/SpladeEmbedder.java

```diff
@@ -167,4 +158,42 @@ public void deconstruct() {
         evaluator.close();
         tokenizer.close();
     }
+
+    /**
+     * Custom max reduce over the sequence dimension of the output tensor. This
+     * to reduce GC pressure from generic Tensor.reduce operation.
+     *
+     * @param tensor the model output tensor of shape d1,dim where d1 is the sequence length and dim is size
+     *               of the vocabulary
+     * @return A tensor of shape d1,1 where each value is the max of the log of the relu of the input tensor
+     */
+    private static IndexedTensor customMaxReduceOverLogOfRelu(IndexedTensor tensor) {
```

Comment:
Consider inlining this logic with for loop in `sparsify()` - the temporary tensor is not really needed.

Reply:
Great feedback!